core/lib/zksync_core/src/sync_layer/sync_state.rs

```diff
@@ -73,6 +77,44 @@ impl SyncState {
     pub(crate) fn is_synced(&self) -> bool {
         self.0.borrow().is_synced().0
     }
+
+    pub async fn run_updater(
+        self,
+        connection_pool: ConnectionPool<Core>,
+        main_node_client: HttpClient,
+        mut stop_receiver: watch::Receiver<bool>,
+    ) -> anyhow::Result<()> {
+        const UPDATE_INTERVAL: Duration = Duration::from_secs(10);
+
+        loop {
+            if *stop_receiver.borrow() {
+                return Ok(());
+            }
```

Comment:
Nit: You can simplify this and the other similar `loop` to

```rust
while !stop_receiver.borrow_and_update() {
    // ...
}
```

`borrow_and_update` is marginally preferable in this scenario because `borrow` doesn't update the latest received value of the `Receiver`; i.e., if hypothetically a no-op update was sent, `stop_receiver.changed()` below would spuriously resolve immediately. OTOH, a no-op update is a not a very realistic scenario.